packages/core/utils/src/dml/helpers/entity-builder/build-indexes.ts

```diff
@@ -42,3 +42,33 @@ function transformWhere(where?: string): string {
 
   return where
 }
+
+/*
+  When a "belongsTo" relationship is found on the DML schema, we create a default index on the
+  foreign key name of the field.
+*/
+export function buildForeignKeyIndexes<TSchema extends DMLSchema>(
+  schema: TSchema
+) {
+  const indexes: EntityIndex<TSchema, QueryCondition>[] = []
+
+  for (const [field, property] of Object.entries(schema)) {
+    const parsed = property.parse(field)
+    const isRelationshipType = "type" in parsed
+    const isBelongsTo = isRelationshipType && parsed.type === "belongsTo"
```

Comment:
Isn't the `parsed.type === "belongsTo"` sufficient?

Reply:
We have a static method on the relationship to handle this check, based on a symbole string we have on the class to be more robust